### PR TITLE
 [MJAVADOC-723] Upgrade Maven Reporting API to 3.1.1/Complete with Maven Reporting Impl 3.2.0

### PR DESCRIPTION
**Pull request by pzygielo**
To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/javadoc/JavadocReport.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 import java.util.ResourceBundle;
 import java.util.stream.Collectors;
 
-import org.apache.maven.doxia.sink.Sink;
+import org.codehaus.doxia.sink.Sink;
```
Comment (olamy) [inline src/main/java/org/apache/maven/plugins/javadoc/JavadocReport.java:30]: ??
I don't think it's a right change. but looks to be back in the past :)

**Review by olamy — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/maven/plugins/javadoc/JavadocReport.java (hunk 1 above)
Comment (pzygielo) [reply src/main/java/org/apache/maven/plugins/javadoc/JavadocReport.java:30]: > ?? I don't think it's a right change. but looks to be back in the past :)

I think that's why MSHARED-1118 has "Restore" in its title
- https://github.com/apache/maven-reporting-api/pull/13

Comment (olamy) [reply src/main/java/org/apache/maven/plugins/javadoc/JavadocReport.java:30]: doh :)

Comment (michael-o) [reply src/main/java/org/apache/maven/plugins/javadoc/JavadocReport.java]: @olamy If you read the reason for the change you'll see that it was my fault to replace this in a minor version. I am working on a followup to this to make it right. Unfortunately, MJAVADOC does not/cannot use maven-reporting-impl which would make this much easier.

**Comment by michael-o** [comment]
Folks, eyes please. This should now do it. Everything passes and it should now be like using m-r-impl. Maven JXR passes with this.

**Review by michael-o — APPROVED** [review]

**Comment by pzygielo** [comment]
I confirm that on my side - plugin with such change (3.4.1-SNAPSHOT a10f0b11e89fd65d61938186ddf16fb86ded239e) works fine when forked by m-site-plugin:3.12.1

Thanks!

**Comment by michael-o** [comment]
Guys, please check. This is a blocker.

**Pull request merged**